packages/base/source/server/errors.js

```diff
@@ -1,22 +1,15 @@
-
-Space.Error.extend(Donations, `AppealNotOpenForNewPledges`, {
-  message: `Appeal not open for new pledges.`
-});
-
-Space.Error.extend(Donations, `AppealNotOpenToAcceptPledge`, {
-  message: `Appeal not open to accept pledge.`
-});
-
-Space.Error.extend(Donations, `AppealNotOpenToDeclinePledge`, {
-  message: `Appeal not open to decline pledge.`
-});
-
-Space.Error.extend(Donations, `AppealNotOpenToWriteOffPledge`, {
-  message: `Appeal not open to write off pledge.`
+Space.Error.extend(Donations, `InvalidAppealState`, {
+  Constructor(commandName, currentState) {
+    Space.Error.constructor.call(this);
+    this.message = `Cannot ${commandName} when in ${currentState} state`;
+  }
 });
 
-Space.Error.extend(Donations, `PledgeHasToBeAcceptedBeforeFulfilled`, {
-  message: `Pledge has to be accepted before it can be fulfilled.`
+Space.Error.extend(Donations, `InvalidPledgeState`, {
+  Constructor(commandName, currentState) {
+    Space.Error.constructor.call(this);
+    this.message = `Cannot ${commandName} when in ${currentState} state`;
+  }
```

Comment:
This could be generalised to all aggregates too (and pushed into the module) if we decide to encapsulate state change rules.